File: src/dataset/c/robot-simulator_exercism-EzequielMarcelo/openAI/ChatGPT4_robot_simulator_v3.c
#include "robot_simulator.h"

void turn_left(robot_status_t *robot);
void turn_right(robot_status_t *robot);
void advance(robot_status_t *robot);

robot_status_t robot_create(robot_direction_t direction, int x, int y) 
{
    return (robot_status_t){.direction = direction, .position = {.x = x, .y = y}};
}

void turn_left(robot_status_t *robot) 
{
    robot->direction = (robot->direction + 3) & 3;
}

void turn_right(robot_status_t *robot) 
{
    robot->direction = (robot->direction + 1) & 3;
}

void advance(robot_status_t *robot) 
{
    static const int dx[] = {0, 1, 0, -1};
    static const int dy[] = {1, 0, -1, 0};
    robot->position.x += dx[robot->direction];
    robot->position.y += dy[robot->direction];
}

void robot_move(robot_status_t *robot, const char *commands) 
{
    while (*commands) 
    {
        switch (*commands++) 
        {
            case 'R': turn_right(robot); break;
            case 'L': turn_left(robot); break;
            case 'A': advance(robot); break;
        }
    }
}

File: src/dataset/python/robot-simulator_exercism-fruit-in/openAI/ChatGPT4_robot_simulator_v3.py
EAST, NORTH, WEST, SOUTH = 1, 0, 3, 2

class Robot:
    def __init__(self, direction=NORTH, x=0, y=0):
        self.coordinates = (x, y)
        self.direction = direction

    def turn_right(self):
        self.direction = (self.direction + 1) % 4

    def turn_left(self):
        self.direction = (self.direction - 1) % 4

    def advance(self):
        x, y = self.coordinates
        moves = {NORTH: (0, 1), EAST: (1, 0), SOUTH: (0, -1), WEST: (-1, 0)}
        dx, dy = moves[self.direction]
        self.coordinates = (x + dx, y + dy)

    def move(self, instructions):
        actions = {
            'R': self.turn_right,
            'L': self.turn_left,
            'A': self.advance
        }
        for ins in instructions:
            actions[ins]()